src/utils/makeReactNativeConfig.js

```diff
@@ -224,13 +230,26 @@ function makeReactNativeConfig(
   const defaultWebpackConfig = getDefaultConfig(env);
   const polyfillPath = require.resolve('./polyfillEnvironment.js');
 
-  const userConfig =
-    typeof userWebpackConfig === 'function'
-      ? userWebpackConfig(env, defaultWebpackConfig)
-      : userWebpackConfig;
+  /**
+  * Load config from file. Support both - "module.exports" and "export default {}"
+  */
+  const { webpack: webpackConfigFactory /* ...haulConfig */ } =
+    userWebpackConfig.default || userWebpackConfig;
+
+  if (
+    typeof webpackConfigFactory !== 'function' ||
+    typeof webpackConfigFactory !== 'string'
```

Comment:
I think this is better: `typeof webpackConfigFactory !== 'function' &&  typeof webpackConfigFactory !== 'object'`

Reply:
Oh, damn, it should be an object. 😅